classes/admin/admin_setting_userselection.php

```diff
@@ -0,0 +1,82 @@
+<?php
+// This file is part of Echo360 Sync
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * @package     logstore_xapi
+ * @author      Heena Agheda <heenaagheda@catalyst-au.net>
+ * @copyright   2021 Catalyst IT Australia {@link http://www.catalyst-au.net}
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace logstore_xapi\admin;
+
+defined('MOODLE_INTERNAL') || die();
+
+global $CFG;
+require_once($CFG->libdir . '/adminlib.php');
+require_once($CFG->libdir . '/form/autocomplete.php');
+
+class admin_setting_userselection extends \admin_setting {
+
+    public function __construct($name, $visiblename, $description, $defaultsetting = []) {
+        parent::__construct($name, $visiblename, $description, $defaultsetting, null);
+    }
+
+     /**
+     * Return the currently selected users.
+     *
+     * @return array The ids of the currently selected users.
+     */
+    public function get_setting() {
+        return explode(',', $this->config_read($this->name));
```

Comment:
Does `get_config('logstore_xapi', 'excludeusers')` end up calling this? If so, then we're double-exploding...

Reply:
No it does not call this...

Reply:
Cool, no worries then. :)